spec/predicates/vuln.md

```diff
@@ -0,0 +1,170 @@
+
+## Prerequisites
+
+The in-toto [attestation] framework and a Vulnerability scanner tools such as [Grype](https://github.com/anchore/grype), [Trivy](https://github.com/aquasecurity/trivy) and others.
+
+## Use cases
+
+When sharing the results of a vulnerability scan using an attestation, there is certain metadata that is crucial to trust and reuse this information.
+Information about the scanner used during the scanning is relevant to trust these results. The state of the vulnerability database used to search for vulnerabilities defines the accuracy of the results. Other metadata information such as the timestamp when the scan finished could define the reusability of these results.
+
+## Model
+
+This is a predicate type that fits within the larger [Attestation] framework.
+The following model aims to provide a well defined list of fields so that consumers know how to start exchanging their scanner results.
+
+This predicate model is inspired by [cosign vulnerability attestation](https://github.com/sigstore/cosign/blob/main/specs/COSIGN_VULN_ATTESTATION_SPEC.md).
+
+## Schema
+
+The schema of this predicate type is documented below.
+
+### Fields
+
+The fields that make up this predicate type are:
+
+The `subject` contains whatever software artifacts are to be associated with this vulnerability report document.
+The `predicate` contains a JSON-encoded data with the following fields:
+
+**scanner** object
+
+> There are lots of container image scanners such as Trivy, Grype, Clair, etc.
+> This field describes which scanner is used while performing a container image scan,
+> as well as version information and which Vulnerability DB is used.
+
+**scanner.uri** string (ResourceURI), optional
+
+> > URI indicating the identity of the source of the scanner.
+
+**scanner.version** string (ResourceURI), optional
+
+> > The version of the scanner.
+
+**scanner.db.uri** string (ResourceURI), optional
+
+> > > URI indicating the identity of the source of the Vulnerability DB.
+
+**scanner.db.version** string, optional
+
+> > > The version of the Vulnerability DB.
+
+**scanner.db.lastUpdate, required** string (timestamp)
+
+> > > The timestamp of when the vulnerability DB was updated last time.
+
+**scanner.result** list
+
+> > The result contains a list of vulnerabilities.
+> > This is the most important part of this field because it'll store the scan result as a whole. So, people might want
+> > to use this field to take decisions based on them by making use of Policy Engines tooling whether allow or deny these images.
+
+**scanner.result.[*].vulnerability** object
+
+> > > The vulnerability object defines information about each one of the vulnerabilities found by the scanner.
+
+**scanner.result.[*].vulnerability.id** string
+
+> > > > This is the identifier of the vulnerability, e.g. GHSA-r9p9-mrjm-926w, CVE-123.
```

Comment:
do we want to include known aliases (list) here for example: https://github.com/advisories/GHSA-fxph-q3j8-mv87 is also CVE-2017-5645

Reply:
It sounds good to me.

Reply:
oh sorry, I think my comment was not specific enough. I was thinking about adding a new field that calls out known aliases (if known). Does that make sense?